app/gui2/shared/languageServerTypes.ts

```diff
@@ -74,6 +74,15 @@ export interface Position {
   character: number
 }
 
+interface IdMapSpan {
+  index: { value: number }
+  size: { value: number }
+}
+
+export type IdMapTuple = [IdMapSpan, string]
+
+export type IdMapTriple = [number, number, string]
```

Comment:
Is this an attempt to make the serialized format of metadata more effective? Are other parts of the system ready for `IdMapTripple`?

Reply:
For now, it is only related to the language server API